parsers/esdtTransferParser.go

```diff
@@ -125,7 +125,9 @@ func (e *esdtTransferParser) parseMultiESDTNFTTransfer(sndAddr, rcvAddr []byte,
 	numOfTransfer := big.NewInt(0).SetBytes(args[0])
 	startIndex := uint64(1)
 	isTxAtSender := false
-	if bytes.Equal(sndAddr, rcvAddr) {
+
+	isFirstArgumentAnAddress := len(args[0]) == len(rcvAddr) && !numOfTransfer.IsUint64()
```

Comment:
Consider using a constant instead of `len(rcvAddr)`. Do we have an `AddressLength = 32` constant somewhere?

Reply:
in a lot of places we are comparing to the receiver address. or tx.sndAddr. We do not really like constants